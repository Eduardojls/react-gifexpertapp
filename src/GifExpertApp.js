import { useState } from "react";
import { AddCategory } from "./component/AddCategory";
import { GifGrid } from "./component/GifGrid";

export const GifExpertApp = () => {
  // Where home screen content is loaded
  const [categories, setCategories] = useState(["shingeki"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
