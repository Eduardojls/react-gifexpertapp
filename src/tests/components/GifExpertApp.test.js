import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    
    test('debe de cargar el componente correctamente', () => {
       
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });
    
});
