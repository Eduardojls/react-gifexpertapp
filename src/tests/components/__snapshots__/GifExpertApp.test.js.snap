// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de cargar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="shingeki"
      key="shingeki"
    />
  </ol>
</Fragment>
`;
